feat(cdns): Added vpn-indexer CLI for Signup Tx Build

arepala-uml opened the pull request:
1. Added vpn-indexer CLI where I have added option to user to provide the refdata.json instead of fetching it from database.
2. Need to implement fetching the refdata from on chain using kupo client address as well ogmios url.

```
akhil-mac@Akhils-MacBook-Air vpn-indexer % go run ./cmd/vpn-cli signup --help
Build an unsigned signup transaction

Usage:
  vpn-cli signup [flags]

Flags:
      --duration int        plan duration in milliseconds
  -h, --help                help for signup
      --kupo-url string     Kupo endpoint (used if --refdata not provided)
      --ogmios-url string   Ogmios endpoint (optional)
      --owner string        owner bech32 address
      --payment string      client payment bech32 address
      --price int           plan price in lovelace
      --region string       region code

Global Flags:
      --format string   output format: hex|cbor (default "hex")
  -o, --out string      output file (default: stdout)
```

wolf31o2 reviewed (changes requested):
I realized that you *must* set a Kupo/Ogmios address, as they're required by the Apollo OgmiosContext we're configuring to fetch the UTxOs for the given user address.

```
go run ./cmd/vpn-cli signup \
--duration 3600000 --price 3000000 --refdata refdata.json \
--payment addr_test1qqynu5d8p9yc7garta6z4g34e2tlzye5ty8uy6ljmffnpnjv7ncp3yppt0gcr50u60y43x32fgadhnl35u9hfqyql2pqepxt0y \
--owner addr_test1qqynu5d8p9yc7garta6z4g34e2tlzye5ty8uy6ljmffnpnjv7ncp3yppt0gcr50u60y43x32fgadhnl35u9hfqyql2pqepxt0y \
--region 'us east-1'
2025/10/16 18:37:01 unable to fetch matches: Get "http://127.0.0.1:1442/v1/matches/addr_test1qqynu5d8p9yc7garta6z4g34e2tlzye5ty8uy6ljmffnpnjv7ncp3yppt0gcr50u60y43x32fgadhnl35u9hfqyql2pqepxt0y?unspent": dial tcp 127.0.0.1:1442: connect: connection refusedOgmiosChainContext: AddressUtxos: kupo request failed
exit status 1
```

There's no reason to support JSON lookups. Kupo is required, so it's always available.

arepala-uml commented:
Removed json lookup part and will be able to fetch the refdata using kupo url only. I was able to successfully parse the kugo reponse and added decoing of dataumHash to get the Prices & regions for refdata. Able to fetch the unsigned transaction hex after doing signup with CLI. 

<img width="1429" height="772" alt="Screenshot 2025-10-19 at 6 20 28 PM" src="https://github.com/user-attachments/assets/239b6d7a-19de-436c-8628-80379ba74487" />


<img width="1429" height="394" alt="Screenshot 2025-10-19 at 6 22 05 PM" src="https://github.com/user-attachments/assets/a747f494-6ff9-4d01-819d-52cf851bd4ae" />
